v3/integrations/nrfasthttp/nrfasthttp_client.go

```diff
@@ -0,0 +1,21 @@
+package nrfasthttp
+
+import (
+	newrelic "github.com/newrelic/go-agent/v3/newrelic"
+	"github.com/valyala/fasthttp"
+)
+
+type FastHTTPClient interface {
```

Comment:
maybe providing a wrapped object like you do for request might be a better design pattern here.